astropy/units/tests/test_units.py

```diff
@@ -128,6 +128,18 @@ def test_units_manipulation():
     (u.AA * u.erg) ** 9
 
 
+def test_compose():
+    x =  u.m.compose(units=(u.m))
+    assert x[0].bases[0] is u.m
+    assert x[0].scale == 1.0
+    x = u.m.compose(units=[u.km], include_prefix_units=True)
+    assert x[0].bases[0] is u.km
+    assert x[0].scale == 0.001
+    x = u.m.compose(units=[u.km])
+    assert x[0].bases[0] is u.km
+    assert x[0].scale == 0.001
+
```

Comment:
Of course, also important to test whether things *fail* when `include_prefix_units=False`:
```
with raises(u.UnitsError):
    (u.km/u.s).compose(units=(u.pc, u.Myr), include_prefix_units=False)
```